server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletGoalState.java

```diff
@@ -0,0 +1,175 @@
+
+    if (state == TabletGoalState.HOSTED) {
+      if (!params.isParentLevelUpgraded()) {
+        // The place where this tablet stores its metadata was not upgraded, so do not assign this
+        // tablet yet.
+        return TabletGoalState.UNASSIGNED;
+      }
+
+      if (tm.getOperationId() != null) {
+        return TabletGoalState.UNASSIGNED;
+      }
+
+      if (!params.isTableOnline(tm.getTableId())) {
+        return TabletGoalState.UNASSIGNED;
+      }
+
+      switch (tm.getHostingGoal()) {
+        case NEVER:
+          return TabletGoalState.UNASSIGNED;
+        case ONDEMAND:
+          if (!tm.getHostingRequested()) {
+            return TabletGoalState.UNASSIGNED;
+          }
+      }
+
+      TServerInstance dest = params.getMigrations().get(extent);
+      if (dest != null && tm.hasCurrent() && !dest.equals(tm.getLocation().getServerInstance())) {
+        return TabletGoalState.UNASSIGNED;
+      }
+
+      if (currentState == TabletState.HOSTED && balancer != null) {
+        // see if the balancer thinks this tablet needs to be unassigned
+
+        Preconditions.checkArgument(
+            tm.getLocation().getType() == TabletMetadata.LocationType.CURRENT,
+            "Expected current tablet location %s %s", tm.getExtent(), tm.getLocation());
+        var tsii = new TabletServerIdImpl(tm.getLocation().getServerInstance());
+
+        var resourceGroup = params.getResourceGroup(tm.getLocation().getServerInstance());
+
+        if (resourceGroup != null) {
+          var reassign = balancer.needsReassignment(new TabletBalancer.CurrentAssignment() {
+            @Override
+            public TabletId getTablet() {
+              return new TabletIdImpl(tm.getExtent());
+            }
+
+            @Override
+            public TabletServerId getTabletServer() {
+              return tsii;
+            }
+
+            @Override
+            public String getResourceGroup() {
+              return resourceGroup;
+            }
+          });
+
+          if (reassign) {
+            return UNASSIGNED;
+          }
+        } else {
+          // A tablet server should always have a resource group, however there is a race
+          // conditions where the resource group map was read before a tablet server came into
+          // existence. Another possible cause for an absent resource group is a bug in accumulo.
+          // In either case do not call the balancer for now with the assumption that the resource
+          // group will be available later. Log a message in case it is a bug.
+          log.trace(
+              "Could not find resource group for tserver {}, so did not consult balancer.  Assuming this is a temporary race condition.",
+              tm.getLocation().getServerInstance());
+        }
+      }
+
+      if (tm.hasCurrent()
+          && params.getServersToShutdown().contains(tm.getLocation().getServerInstance())) {
+        return TabletGoalState.SUSPENDED;
```

Comment:
#3896 was not merged, but one of the changes I made in there was:
```
       if (tgw.canSuspendTablets()) {
          return TabletGoalState.SUSPENDED;
        } else {
          return TabletGoalState.UNASSIGNED;
        }
```
This is because the TGW for the Root table returns false for `canSuspendTablets`.

Reply:
done in 1f10135